### PR TITLE
Add login prompt refactor issue 3847

### PR DESCRIPTION
**Pull request by gyi2521**
Fixes #3847 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/views/questions/_questions.html.erb [hunk 1]
```diff
@@ -1,6 +1,7 @@
+<% notes = questions || @questions # accept local if present, default to instance %>
```
Comment (jywarren) [inline app/views/questions/_questions.html.erb]: Hmm, @cesswairimu i'm seeing this in the errors... maybe we have to be sure this is set upstream? 

```
ActionView::Template::Error:         ActionView::Template::Error: undefined local variable or method `questions' for #<#<Class:0x000055b41d5a1b18>:0x000055b41dbc7538>
        Did you mean?  @questions
            app/views/questions/_questions.html.erb:1:in 
```

Comment (gyi2521) [reply app/views/questions/_questions.html.erb]: Oops , it should be <% questions = questions || @questions # accept local if present, default to instance %> ... Is it okay to change it here?

Comment (cesswairimu) [reply app/views/questions/_questions.html.erb]: @gyi2521 yes you can change it here and commit. @jywarren yes it should be best to set upstream and origin to their normal forks.
@gyi2521 you can do this by running this commands
`git remote remove origin`

Comment (gyi2521) [reply app/views/questions/_questions.html.erb]: Hi @cesswairimu , the only part I merged is that when I made the second PR , it said  I had a conflict so resolved the conflict by deleting my code... The conflicted code was as following:
<div class="text-center"><%= will_paginate wikis, :renderer => BootstrapPagination::Rails if @paginated %></div> ... on app/views/wiki/_wikis.html.erb page.  Somehow,<div class="text-center"> didn't exist on the version I worked on... but everything should look okay now.

@jywarren I changed the <% notes = questions || @questions > to <% questions = questions || @questions> just now, and it looks like ALL checks have passed.  Is it ready to be merged FINALLY?  :-) 

Thank you both for your support!

**Comment by jywarren** [comment]
This looks super. Thanks so much!!!

**Comment by jywarren** [comment]
and thanks @cesswairimu for the teamwork here!!!

**Comment by gyi2521** [comment]
Thank you all!

**Pull request merged**